Yoast/wordpress-seo
# Use tailwind combobox as user-select dropdown

* HeadlessUI / TailwindUI released a nicely styled combobox that ALMOST covers the use cases we have with our wordpress user select. So let's use it!
* The exception is that the Combobox opens slightly differently (yet accessible), and that it is not async, like our current implementation (which is an async react-select from a package)

## Summary


This PR can be summarized in the following changelog entry:

* Uses a HeadlessUI combobox instead of an async react-select component for the first time configuration's user select.

## Relevant technical choices:

* Created an async wrapper for the base combobox component
* ~Implemented additional UX requirements with the exception of opening on click input, since that gave me some issues with aria-expanded booleans that were not within my control. Should pick up in a separate issue (DUPP 321) or create an issue with HeadlessUI.~ fixed here after all.
* We now return a saved user name from the PHP since it seemed silly to query this in "JS' time" while we have easy access in "PHP time".

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* The UserSelect in the person section should work as intended ™️ ... Note that in the design it is not a combobox. Rather, perhaps you can compare in functionality with the user select in the search-appearance settings (though of course we have different styling here).
* Select "Person" in the site representation step. If it is empty it should display a placeholder. (Devs can fake this by making this function in `configuration-workout-integration` (line 276) return false.)
```php
	/**
	 * Gets the person id from the option in the database.
	 *
	 * @return int|null The person id, null if empty.
	 */
	private function get_person_id() {
		return $this->options_helper->get( 'company_or_person_user_id' );
	}
```
* If you return false above to test the placeholder: undo those changes.
* Test whether saving users works.
* Verify that the alert appears upon user selection.
* Verify the options box opens and closes as expected.
* Verify the options box filters as expected when changing the input.
* Verify the selected option is highlighted in the dropdown.
* Verify clicking the input opens the options, and clicking outside closes it again (and refills the users).
* Verify keyboard navigation works (tabbing to the input field, using arrow up and down to navigate options, enter to submit, esc to close, etc.)

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* * The UserSelect in the person section should work as intended ™️ ... Note that in the design it is not a combobox. Rather, perhaps you can compare in functionality with the user select in the search-appearance settings (though of course we have different styling here).
* Select "Person" in the site representation step. If it is empty it should display a placeholder. (not sure QA can fake this, but perhaps with the factory reset it works).
* If you return false above to test the placeholder: undo those changes.
* Test whether saving users works.
* Verify that the alert appears upon user selection.
* Verify the options box opens and closes as expected.
* Verify the options box filters as expected when changing the input.
* Verify the selected option is highlighted in the dropdown.
* * Verify clicking the input opens the options, and clicking outside closes it again (and refills the users).
* Verify keyboard navigation works (tabbing to the input field, using arrow up and down to navigate options, enter to submit, esc to close, etc.)

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes DUPP-288
Fixes DUPP-321
